### PR TITLE
feat: upgrade to sarknet.js v6.11.0

### PR DESCRIPTION
**Pull request by khanti42**
Closes issue #277

### CODE REVIEW

**Review by stanleyyconsensys — CHANGES_REQUESTED** [review]
Can we separate the http-server in other PR?
beside what is the reason to have the debug UI? it is quite outdate
and please also update the starknet.js in wallet ui,get-starknet too

File: packages/starknet-snap/src/utils/keyPair.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import type { BIP44AddressKeyDeriver } from '@metamask/key-tree';
 import { getBIP44AddressKeyDeriver } from '@metamask/key-tree';
 import { utils } from 'ethers';
-import { number, ec } from 'starknet_v4.22.0';
+import { num as numUtils, ec } from 'starknet';
```
Comment (stanleyyconsensys) [inline packages/starknet-snap/src/utils/keyPair.ts:4]: this is very risky change, are we confirm the address is always the same as before

File: packages/wallet-ui/package.json [hunk 2]
```diff
@@ -6,19 +6,19 @@
   "license": "(Apache-2.0 OR MIT)",
   "scripts": {
     "allow-scripts": "yarn workspace root allow-scripts",
-    "build": "react-scripts build",
+    "build": "craco build",
```
Comment (stanleyyconsensys) [inline packages/wallet-ui/package.json:9]: what is the reason to using craco? when every other dapp in consensys is using react-scripts / gasby

**Review by stanleyyconsensys — CHANGES_REQUESTED** [review]

File: packages/wallet-ui/package.json (hunk 2 above)
Comment (stanleyyconsensys) [reply packages/wallet-ui/package.json:9]: get-starknet still using 5.14.0

File: packages/starknet-snap/src/utils/keyPair.ts (hunk 1 above)
Comment (khanti42) [reply packages/starknet-snap/src/utils/keyPair.ts]: Ok, because of potential big implication we rollback this and will be done later on an other PR

File: packages/wallet-ui/package.json (hunk 2 above)
Comment (stanleyyconsensys) [inline packages/wallet-ui/package.json:9]: Move craco to another PR

**Review by stanleyyconsensys — CHANGES_REQUESTED** [review]

File: packages/wallet-ui/craco.config.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+module.exports = {
```
Comment (stanleyyconsensys) [inline packages/wallet-ui/craco.config.js:1]: Craco is not removed

File: yarn.lock [hunk 4]
```diff
@@ -2130,13 +2130,32 @@ __metadata:
...
     sinon-chai: ^3.7.0
-    starknet: 6.7.0
+    starknet: ^6.11.0
     starknet_v4.22.0: "npm:starknet@4.22.0"
     ts-node: ^10.9.2
     typescript: ^4.7.4
   languageName: unknown
   linkType: soft
 
+"@craco/craco@npm:^7.1.0":
```
Comment (stanleyyconsensys) [inline yarn.lock:2140]: craco is not removed

**Review by stanleyyconsensys — CHANGES_REQUESTED** [review]

File: packages/get-starknet/src/snap.ts [hunk 5]
```diff
@@ -109,7 +110,7 @@ export class MetaMaskSnap {
     senderAddress: string,
     txnInvocation: AllowArray<Call>,
     abis?: Abi[],
-    invocationsDetails?: InvocationsDetails,
+    invocationsDetails?: UniversalDetails,
   ): Promise<InvokeFunctionResponse> {
```
Comment (stanleyyconsensys) [inline packages/get-starknet/src/snap.ts:114]: it looks like it is not sync with the snap

**Review by stanleyyconsensys — CHANGES_REQUESTED** [review]

File: packages/get-starknet/src/snap.ts (hunk 5 above)
Comment (stanleyyconsensys) [reply packages/get-starknet/src/snap.ts:114]: we keep the snap in sync with invocationsDetails for now and fix in the other PR

**Comment by stanleyyconsensys** [comment]
Can we pin the version in this PR?

**Review by stanleyyconsensys — DISMISSED** [review]
LGTM

**Review by stanleyyconsensys — APPROVED** [review]
